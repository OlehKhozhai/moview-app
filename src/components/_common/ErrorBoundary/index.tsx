import React from 'react';

import Button from 'components/_common/Button';
import styles from './styles.module.scss';

type ErrorBoundaryProps = {
  children: React.ReactNode;
};

type ErrorBoundaryState = {
  hasError: boolean;
};

export default class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError() {
    return { hasError: true };
  }

  render() {
    if (this.state.hasError) {
      return (
        <div className={styles.root}>
          <h1 className={styles.title}>Oops, something went wrong</h1>
          <Button title="Home" />
        </div>
      );
    }

    return this.props.children;
  }
}
