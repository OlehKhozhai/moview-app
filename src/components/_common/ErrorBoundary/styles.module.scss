.root {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-flow: column;
  margin-top: 100px;
}

.title {
  font-size: 30px;
  margin-bottom: 20px;
}
