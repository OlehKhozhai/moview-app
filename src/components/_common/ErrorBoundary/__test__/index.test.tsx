import React from 'react';
import { create } from 'react-test-renderer';

import ErrorBoundary from '../index';

const ProblemChild = () => {
  throw new Error('Error thrown from problem child');
  return <div>Error</div>;
};

describe('ErrorBoundary component', () => {
  test('should render ErrorBoundary component without error', () => {
    const component = create(
      <ErrorBoundary>
        <div>Test children</div>
      </ErrorBoundary>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render ErrorBoundary component with error', () => {
    // ToDo fix error message in terminal during testing
    const component = create(
      <ErrorBoundary>
        <ProblemChild />
      </ErrorBoundary>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
