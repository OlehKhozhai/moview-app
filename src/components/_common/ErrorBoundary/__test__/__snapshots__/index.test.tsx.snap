// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should render ErrorBoundary component with error 1`] = `
<div
  className="root"
>
  <h1
    className="title"
  >
    Oops, something went wrong
  </h1>
  <button
    className="root primary"
    disabled={false}
    type="button"
  >
    Home
  </button>
</div>
`;

exports[`ErrorBoundary component should render ErrorBoundary component without error 1`] = `
<div>
  Test children
</div>
`;
